### PR TITLE
Update to NixOS 23.11

### PR DESCRIPTION
**Pull request by erikarvstedt**
Todo:
- [x] Fix Python-based pkgs like clightning plugins and joinmarket
- [x] Manual testing

### CODE REVIEW

**Comment by erikarvstedt** [comment]
@nixbitcoin, could you do some joinmarket testing?

File: pkgs/python-packages/jmclient/default.nix [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
-      --replace "'pyjwt==2.4.0'" "'pyjwt==2.6.0'"
+      --replace "'argon2_cffi==21.3.0'" "'argon2_cffi==23.1.0'"
+    substituteInPlace setup.py \
+      --replace "'pyjwt==2.4.0'" "'pyjwt==2.8.0'"
+    substituteInPlace setup.py \
+      --replace "'werkzeug==2.2.3'" "'werkzeug==2.3.7'"
   '';
 
+  # The unit tests can't be run in a Nix build environment
+  doCheck = false;
```
Comment (jonasnick) [inline pkgs/python-packages/jmclient/default.nix:55]: If I set this to true the build keeps succeeding. Do you mean that the tests that do run are unnecessary?

File: pkgs/build-support/fetch-node-modules.nix [hunk 2]
```diff
@@ -25,6 +25,9 @@ stdenvNoCC.mkDerivation ({
 
   phases = "unpackPhase patchPhase buildPhase installPhase";
 
+  # npm doesn't support var `SSL_CERT_FILE`.
+  NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
```
Comment (jonasnick) [inline pkgs/build-support/fetch-node-modules.nix:29]: How did you find out about this CA issue? https://github.com/NixOS/nixpkgs/pull/257513 suggests that npm has hardcoded CAs.

File: pkgs/python-packages/default.nix [hunk 3]
```diff
@@ -5,7 +5,7 @@ let
   # be supported through the NixOS 23.05 release cycle.`
   # TODO-EXTERNAL: consider removing when
   # https://github.com/Simplexum/python-bitcointx/issues/76 and
-  # https://github.com/JoinMarket-Org/joinmarket-clientserver#1451 are resolved.
+  # https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/1451 are resolved.
```
Comment (jonasnick) [inline pkgs/python-packages/default.nix:8]: How about
```
consider removing when a fix to
https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/1613
has been released.
```

**Review by jonasnick — COMMENTED** [review]
We can also update `install.md` (maybe mention in `dev/README.md` "Switching to a new NixOS release" as well):   
```  
wget  https://releases.nixos.org/nixos/23.11/nixos-23.11.1494.b4372c4924d9/nixos-minimal-23.11.1494.b4372c4924d9-x86_64-linux.iso
sha256sum nixos-minimal-23.11.1494.b4372c4924d9-x86_64-linux.iso
# output: f48cf810432c1f04b291c947b36f824823dfef8ebfa0e1906602a516450189d8
```

File: pkgs/python-packages/jmclient/default.nix (hunk 1 above)
Comment (erikarvstedt) [reply pkgs/python-packages/jmclient/default.nix:55]: With `doCheck = true`, the logs show `Ran 0 tests in 0.000s`. Running the tests requires adding `pytestCheckHook`, which then shows that the tests hang. (Probably while setting up a bitcoind regtest instance. I haven't looked into this closely.)

File: pkgs/build-support/fetch-node-modules.nix (hunk 2 above)
Comment (erikarvstedt) [reply pkgs/build-support/fetch-node-modules.nix:29]: Without `NODE_EXTRA_CA_CERTS`, the `npm ci` command in `fetchNodeModules` hangs.
You can repro this by setting the hash in `backend = fetchNodeModules` to an empty string and then running `nix build --no-link -L .#mempool-backend`.

**Comment by erikarvstedt** [comment]
- I've updated this PR to the latest 23.11 channel update.
- I've added `python-bitcointx: 1.1.3 -> 1.1.4` which removes the openssl 1.1 dependency and is compatible with the latest joinmarket release.

Running live on nixbitcoin.org.

File: pkgs/python-packages/default.nix (hunk 3 above)
Comment (erikarvstedt) [reply pkgs/python-packages/default.nix:8]: Resolved by the `python-bitcointx` update (see post below).

Comment (jonasnick) [reply pkgs/python-packages/default.nix:8]: This comment block doesn't seem to be updated?

**Review by jonasnick — COMMENTED** [review]
mu-nit: if you touch the commits again, can you add `clightning: 23.08.1 -> 23.11` to the NixOS-update-commit description?

File: pkgs/build-support/fetch-node-modules.nix (hunk 2 above)
Comment (jonasnick) [reply pkgs/build-support/fetch-node-modules.nix:29]: Hm ok, good that we discovered this!

**Comment by erikarvstedt** [comment]
> mu-nit: if you touch the commits again, can you add clightning: 23.08.1 -> 23.11 to the NixOS-update-commit description?

Yes, I always add pkg updates to the description.
This PR doesn't contain a clightning update: `nix eval .#modulesPkgs.clightning.version` = `23.08.1`.

**Comment by erikarvstedt** [comment]
> This comment block doesn't seem to be updated? 

Thanks, fixed.

**Comment by erikarvstedt** [comment]
I've fixed up a few commit descriptions. No src changes.

**Review by jonasnick — APPROVED** [review]
> This PR doesn't contain a clightning update: nix eval .#modulesPkgs.clightning.version = 23.08.1.

Ah, you only updated the 23.11 channel.

ACK 85bbdb857a40fbe6fa34e9c095112b494e599501

Thanks @erikarvstedt and also @kristapsk for updating joinmarket!

**Pull request merged**